src/app/modules/main/activity_center/model.nim

```diff
@@ -156,27 +156,58 @@ QtObject:
     self.activityCenterNotifications = activityCenterNotifications
     self.endResetModel()
 
-  proc addActivityNotificationItemToList*(self: Model, activityCenterNotification: Item, addToCount: bool = true) =
-    let modelIndex = newQModelIndex()
-    defer: modelIndex.delete
-    self.beginInsertRows(modelIndex, 0, 0)
-    self.activityCenterNotifications.insert(activityCenterNotification, 0)
+  proc updateActivityCenterNotification*(self: Model, ind: int, newNotification: Item) =
+    self.activityCenterNotifications[ind] = newNotification
+    let index = self.createIndex(ind, 0, nil)
+    defer: index.delete
+    self.dataChanged(index, index, @[
```

Comment:
You can now use `self.dataChanged(index, index)` to signal all role changes; you don't have to enum all the roles

Reply:
That's really cool! Fixed

Reply:
Perhaps I didn't advertise it enough :) 

https://github.com/status-im/nimqml/commit/90f19ac030c28bbb4f89bf014afba344ca2941cb

Reply:
And https://github.com/status-im/status-desktop/pull/12102/files